core/src/main/java/org/mskcc/oncotree/OncoTreePersistentCache.java

```diff
@@ -0,0 +1,289 @@
+
+    @Autowired
+    private CachingProvider cachingProvider;
+
+    @Autowired
+    private OncoTreeRepository oncoTreeRepository;
+
+    @Autowired
+    private OncoTreeVersionRepository oncoTreeVersionRepository;
+
+    @Autowired
+    private CrosswalkRepository crosswalkRepository;
+
+    private static final String ONCOTREE_NODES_CACHE = "oncoTreeNodesEHCache";
+    private static final String ONCOTREE_VERSIONS_CACHE = "oncoTreeVersionsEHCache";
+    private static final String MSKCONCEPT_CACHE = "mskConceptEHCache";
+    public static final String ONCOTREE_VERSIONS_CACHE_KEY = "ONCOTREE_VERSIONS_CACHE_KEY";
+    private static final String BACKUP_CACHE_CONFIG_FILENAME = "ehcache_backup.xml";
+
+    public CacheManager getCacheManager(String ehcacheXMLFilename) throws Exception {
+        CacheManager cacheManager = cachingProvider.getCacheManager(getClass().getClassLoader().getResource(ehcacheXMLFilename).toURI(), getClass().getClassLoader());
+        return cacheManager;
+    }
+
+    // retrieve cached TopBraid responses from default EHCache location
+    @Cacheable(value = "oncoTreeNodesEHCache", key = "#version.version", unless = "#result==null")
+    public ArrayList<OncoTreeNode> getOncoTreeNodesFromPersistentCache(Version version) {
+        ArrayList<OncoTreeNode> oncoTreeNodes = new ArrayList<OncoTreeNode>();
+        try {
+            oncoTreeNodes = oncoTreeRepository.getOncoTree(version);
+        } catch (TopBraidException e) {
+            logger.error("Unable to get oncotree nodes from TopBraid... attempting to read from backup.");
+            try {
+                oncoTreeNodes = getOncoTreeNodesFromPersistentCacheBackup(version);
+                if (oncoTreeNodes == null) {
+                    logger.error("No oncotree nodes were found in backup.");
+                    throw new RuntimeException();
+                } 
+            } catch (Exception e2) {
+                logger.error("Unable to read oncotree nodes from backup.");
+                throw new RuntimeException();
+            }
+        }
+        return oncoTreeNodes;
+    }
+
+    @Cacheable(value = "oncoTreeVersionsEHCache", key = "#root.target.ONCOTREE_VERSIONS_CACHE_KEY", unless = "#result==null")
+    public ArrayList<Version> getOncoTreeVersionsFromPersistentCache() {
+        ArrayList<Version> versions = new ArrayList<Version>();
+        try {
+            versions = oncoTreeVersionRepository.getOncoTreeVersions();
+        } catch (TopBraidException e) {
+            logger.error("Unable to get versions from TopBraid... attempting to read from backup.");
+            try {
+                versions = getOncoTreeVersionsFromPersistentCacheBackup();
+                if (versions == null) {
+                    logger.error("No versions were found in backup.");
+                    throw new RuntimeException();
+                }
+            } catch (Exception e2) {
```

Comment:
same comment here regarding the try/catch block structures and throwing RuntimeException

Reply:
same as above